### PR TITLE
Add doc for faucet

### PR DESCRIPTION
**Pull request by 666lcz**

### CODE REVIEW

File: doc/src/build/faucet.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
+title: Requesting Gas Tokens from Faucet
+---
+
+Sui faucet is a helpful tool where Sui developers can get free test SUI tokens to deploy and interact with their programs on Sui's DevNet and TestNet.
```
Comment (ronny-mysten) [inline doc/src/build/faucet.md:5]:
```suggestion
Sui faucet is a helpful tool where Sui developers can get free test SUI tokens to deploy and interact with their programs on Sui's Devnet and Testnet.
```

File: doc/src/build/faucet.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+Test tokens can be requested in the following ways:
```
Comment (ronny-mysten) [inline doc/src/build/faucet.md:7]:
```suggestion
You can request test tokens in the following ways:
```

File: doc/src/build/faucet.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+## Pre-requisites
```
Comment (ronny-mysten) [inline doc/src/build/faucet.md:9]:
```suggestion
## Prerequisites
```

File: doc/src/build/faucet.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+To request tokens from the faucet, one must own a wallet address that can receive the SUI tokens. An address can be generated via the [Sui CLI tool](https://docs.sui.io/devnet/build/cli-client#active-address) or the [Sui wallet](https://docs.sui.io/devnet/explore/wallet-browser).
```
Comment (ronny-mysten) [inline doc/src/build/faucet.md:11]:
```suggestion
To request tokens from the faucet, you must own a wallet address that can receive the SUI tokens. You can generate an address via the [Sui CLI tool](../build/cli-client.md#active-address) or the [Sui wallet](../explore/wallet-browser.md).
```

File: doc/src/build/faucet.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+## 1. Request test tokens through Discord
+
+1. Join [Discord](https://discord.gg/sui).
+   If you try to join the Sui Discord channel using a newly created Discord account you may need to wait a few days for validation.
```
Comment (ronny-mysten) [inline doc/src/build/faucet.md:16]:
```suggestion
   If you try to join the Sui Discord channel using a newly created Discord account, you may need to wait a few days for validation.
```

File: doc/src/build/faucet.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+1. Request test SUI tokens in the Sui [#devnet-faucet](https://discord.com/channels/916379725201563759/971488439931392130) or [#testnet-faucet](https://discord.com/channels/916379725201563759/1037811694564560966) Discord channel. Note that the TestNet faucet is only available when TestNet is live.
```
Comment (ronny-mysten) [inline doc/src/build/faucet.md:17]:
```suggestion
1. Request test SUI tokens in the Sui [#devnet-faucet](https://discord.com/channels/916379725201563759/971488439931392130) or [#testnet-faucet](https://discord.com/channels/916379725201563759/1037811694564560966) Discord channels. Note that the Testnet faucet is only available while Testnet is live.
```

File: doc/src/build/faucet.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+   Send the following message to the channel with your client address:
+   !faucet <Your client address>
```
Comment (ronny-mysten) [inline doc/src/build/faucet.md:19]:
```suggestion
   `!faucet <Your client address>`
```

File: doc/src/build/faucet.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+```
+import { JsonRpcProvider, Network } from '@mysten/sui.js';
+// connect to Devnet
+const provider = new JsonRpcProvider(Network.DEVNET);
+// get tokens from the DevNet faucet server
+await provider.requestSuiFromFaucet(
+  '<YOUR SUI ADDRESS>'
+);
+```
+
```
Comment (666lcz) [inline doc/src/build/faucet.md:59]:
```suggestion

Note: This option will be disabled for TestNet in TestNet Wave 2. Please use Discord channel instead for TestNet wave 2.
```

File: doc/src/build/faucet.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+1. Join [Discord](https://discord.gg/sui).
+   If you try to join the Sui Discord channel using a newly created Discord account you may need to wait a few days for validation.
+1. Request test SUI tokens in the Sui [#devnet-faucet](https://discord.com/channels/916379725201563759/971488439931392130) or [#testnet-faucet](https://discord.com/channels/916379725201563759/1037811694564560966) Discord channel. Note that the TestNet faucet is only available when TestNet is live.
+   Send the following message to the channel with your client address:
+   !faucet <Your client address>
+
+## 2. Request test tokens through wallet
+
+You can request test tokens within the [Sui wallet](https://docs.sui.io/devnet/explore/wallet-browser#add-sui-tokens-to-your-sui-wallet).
```
Comment (ronny-mysten) [inline doc/src/build/faucet.md:23]:
```suggestion
You can request test tokens within [Sui Wallet](../explore/wallet-browser.md#add-sui-tokens-to-your-sui-wallet).
```

File: doc/src/build/faucet.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+Note: This option will be disabled for TestNet in TestNet Wave 2. Please use Discord channel instead for TestNet Wave 2.
```
Comment (ronny-mysten) [inline doc/src/build/faucet.md:25]:
```suggestion
**Important:** This option is disabled for Testnet in Testnet Wave 2. Use the Discord channels instead for Testnet Wave 2.
```

File: doc/src/build/faucet.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+
+## 3. Request test tokens through Curl
```
Comment (ronny-mysten) [inline doc/src/build/faucet.md:27]:
```suggestion
## 3. Request test tokens through cURL
```

File: doc/src/build/faucet.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+You can also use the following Curl command to request tokens directly from the faucet server.
```
Comment (ronny-mysten) [inline doc/src/build/faucet.md:29]:
```suggestion
Use the following cURL command to request tokens directly from the faucet server:
```

File: doc/src/build/faucet.md [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+--header 'Content-Type: application/json' \
+--header 'Cookie: __cf_bm=DZ4EG6GULlrwnyZMGoqwFjD8p6trJzWsY0LxvHd.NJs-1674598151-0-ARjqPuQjq1efkiQX6ItAI/4QejXhgHfA5rgr8oNoKiRskODMvTraH7VHGx7PF7IjgvEJTbIRB52Yia/Z6UfVlpo=; _cfuvid=hwMbc_CMbJrDSx2dM9tblANlIrpdLoCGFTbOdAhl4HM-1674172742724-0-604800000' \
+--data-raw '{
+    "FixedAmountRequest": {
+        "recipient": "<YOUR SUI ADDRESS>"
+    }
+}'
+```
+
+Replace `'https://faucet.devnet.sui.io/gas'` with `http://127.0.0.1:5003/gas` when working with local network.
```
Comment (ronny-mysten) [inline doc/src/build/faucet.md:42]:
```suggestion
Replace `'https://faucet.devnet.sui.io/gas'` with `http://127.0.0.1:5003/gas` when working with a local network.
```

File: doc/src/build/faucet.md [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+
+Note: This option will be disabled for TestNet in TestNet Wave 2. Please use Discord channel instead for TestNet wave 2.
```
Comment (ronny-mysten) [inline doc/src/build/faucet.md:44]:
```suggestion
**Important:** This option is disabled for Testnet in Testnet Wave 2. Use the Discord channels instead for Testnet Wave 2.
```

File: doc/src/build/faucet.md [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+
+## 4. Request test tokens through TypeScript SDK
+
+You can also access the faucet through the TS-SDK.
+
+```
+import { JsonRpcProvider, Network } from '@mysten/sui.js';
+// connect to Devnet
+const provider = new JsonRpcProvider(Network.DEVNET);
+// get tokens from the DevNet faucet server
```
Comment (ronny-mysten) [inline doc/src/build/faucet.md:54]:
```suggestion
// get tokens from the Devnet faucet server
```

File: doc/src/build/faucet.md [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+await provider.requestSuiFromFaucet(
+  '<YOUR SUI ADDRESS>'
+);
+```
+
+Note: This option will be disabled for TestNet in TestNet Wave 2. Please use Discord channel instead for TestNet wave 2.
```
Comment (ronny-mysten) [inline doc/src/build/faucet.md:60]:
```suggestion
**Important:** This option is disabled for Testnet in Testnet Wave 2. Use the Discord channels instead for Testnet Wave 2.
```

**Review by ronny-mysten — COMMENTED** [review]
Just some style guide things and linking for github (also works in site).

**Review by ronny-mysten — APPROVED** [review]

**Pull request merged**